python.d/go_expvar.chart.py

```diff
@@ -0,0 +1,225 @@
+# update_every = 2
+priority = 60000
+retries = 60
+
+
+MEMSTATS_CHARTS = {
+    'memstats.heap': {
+        'options': ['heap', 'memory: size of heap memory structures', 'kB', 'memstats', 'expvar.memstats.heap', 'line'],
+        'lines': [
+            ['memstats.heap_alloc', 'alloc', 'absolute', 1, 1024],
+            ['memstats.heap_inuse', 'inuse', 'absolute', 1, 1024]
+        ]},
+    'memstats.stack': {
+        'options': ['stack', 'memory: size of stack memory structures', 'kB', 'memstats', 'expvar.memstats.stack', 'line'],
+        'lines': [
+            ['memstats.stack_inuse', 'inuse', 'absolute', 1, 1024]
+        ]},
+    'memstats.mspan': {
+        'options': ['mspan', 'memory: size of mspan memory structures', 'kB', 'memstats', 'expvar.memstats.mspan', 'line'],
+        'lines': [
+            ['memstats.mspan_inuse', 'inuse', 'absolute', 1, 1024]
+        ]},
+    'memstats.mcache': {
+        'options': ['mcache', 'memory: size of mcache memory structures', 'kB', 'memstats', 'expvar.memstats.mcache', 'line'],
+        'lines': [
+            ['memstats.mcache_inuse', 'inuse', 'absolute', 1, 1024]
+        ]},
+    'memstats.live_objects': {
+        'options': ['live_objects', 'memory: number of live objects', 'objects', 'memstats', 'expvar.memstats.live_objects', 'line'],
+        'lines': [
+            ['memstats.live_objects', 'live']
+        ]},
+    'memstats.sys': {
+        'options': ['sys', 'memory: size of reserved virtual address space', 'kB', 'memstats', 'expvar.memstats.sys', 'line'],
+        'lines': [
+            ['memstats.sys', 'sys', 'absolute', 1, 1024]
+        ]},
+    'memstats.gc_pauses': {
+        'options': ['gc_pauses', 'memory: average duration of GC pauses', 'ns', 'memstats', 'expvar.memstats.gc_pauses', 'line'],
+        'lines': [
+            ['memstats.gc_pauses', 'avg']
+        ]},
+}
+
+MEMSTATS_ORDER = ['memstats.heap', 'memstats.stack', 'memstats.mspan', 'memstats.mcache', 'memstats.sys', 'memstats.live_objects', 'memstats.gc_pauses']
+
+
+def flatten(d, top='', sep='.'):
+    items = []
+    for key, val in d.items():
+        nkey = top + sep + key if top else key
+        if isinstance(val, dict):
+            items.extend(flatten(val, nkey, sep=sep).items())
+        else:
+            items.append((nkey, val))
+    return dict(items)
+
+
+class Service(UrlService):
+    def __init__(self, configuration=None, name=None):
+        UrlService.__init__(self, configuration=configuration, name=name)
+
+        # if memstats collection is enabled, add the charts and their order
+        if self.configuration.get('collect_memstats'):
+            self.definitions = MEMSTATS_CHARTS
+            self.order = MEMSTATS_ORDER
+        else:
+            self.definitions = dict()
+            self.order = list()
+
+        # if extra charts are defined, parse their config
+        extra_charts = self.configuration.get('extra_charts', dict())
+        if len(extra_charts) > 0:
```

Comment:
Sorry, I have been writing Go code for a while, looks like I forgot a few python idioms :)

Fixed in [d915aa6](https://github.com/firehol/netdata/pull/2280/commits/d915aa62cb6145142876569d6a919e4bee794c15)